### PR TITLE
Add SmallRye Stork intro post

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
** If you are updating a guide, please submit your pull request to the main repository: https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc **

### CODE REVIEW

File: _posts/2021-11-23-smallrye-stork-intro.adoc [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+layout: post
+title: 'SmallRye Stork meets Quarkus'
+date: 2021-11-23
+tags: extension smallrye-stork
```
Comment (cescoffier) [inline _posts/2021-11-23-smallrye-stork-intro.adoc:5]: Aren't tags comma-separated?

File: _posts/2021-11-23-smallrye-stork-intro.adoc [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+synopsis: Introduction to customizable service discovery and client side load balancing with Quarkus and SmallRye Stork
```
Comment (cescoffier) [inline _posts/2021-11-23-smallrye-stork-intro.adoc:6]: client-side and load-balancing

File: _posts/2021-11-23-smallrye-stork-intro.adoc [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+author: michalszynkiewicz
+---
+:imagesdir: /assets/images/posts/smallrye-stork
+
+Last week, the Quarkus Insights spotlighted a new project called SmallRye Stork and its integration in Quarkus. In this blog post, we're going to give you a concise introduction to Stork.
```
Comment (cescoffier) [inline _posts/2021-11-23-smallrye-stork-intro.adoc:11]: Add a link for the insights.

Comment (cescoffier) [inline _posts/2021-11-23-smallrye-stork-intro.adoc:11]: Link for stork

File: _posts/2021-11-23-smallrye-stork-intro.adoc [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+== What is Stork?
+SmallRye Stork is a _service discovery and client-side load-balancing framework_.
+
+Out of the box, it provides integrations with Hashicorp Consul, Eureka, and Kubernetes and brings in a couple of load-balancer (handling the service selection) implementations, such as round-robin. But the most significant strength of Stork is its extensibility. You can plug your own service discovery mechanism or implement your own, maybe business-related, service selection strategy.
+
+== Stork and Quarkus
+As of Quarkus 2.5.0.Final, Quarkus gRPC and the Quarkus Reactive REST Client can use Stork to handle the discovery and the selection of the services.
+
+You just need to configure how you want to locate and select your services and add the related implementation, and the rest will be handled by the clients directly.
```
Comment (cescoffier) [inline _posts/2021-11-23-smallrye-stork-intro.adoc:36]: Remove and add the relevant implementations - too many and

File: _posts/2021-11-23-smallrye-stork-intro.adoc [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+For example, to use Stork to determine the actual address for REST Client Reactive, set the scheme of your client's URI to `stork`, and the hostname of the URI to the name of the Stork service. If your client's `configKey` is `my-client`, to use it with Hashicorp Consul and the `round-robin` load balancer, put the following in your `application.properties:
+
+[source,property]
+.application.properties
+----
+quarkus.rest-client.my-client.uri=stork://hello-service
+
```
Comment (cescoffier) [inline _posts/2021-11-23-smallrye-stork-intro.adoc:44]: Add something like
# configure the discovery and selection of the hello-service

File: _posts/2021-11-23-smallrye-stork-intro.adoc [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+----
+
+The http://smallrye.io/smallrye-stork/1.0.0.Beta1/quarkus/[project's documentation] describes step by step how to use Stork with the REST Client, using Consul and round-robin load balancer as examples.
+
+Service selection can be pretty advanced. For instance, the least-response-time strategy monitors the service calls to select the _fastest_ service instance.
+
+NOTE: In Quarkus 2.5, the gRPC extension does not propagate statistics of the calls, such as response time or encountered errors. Therefore, a load balancer that relies on this data, such as least-response-time, will not work correctly. We are working on it and hope to have it in Quarkus 2.6.
+
+
+=== Demo and a lot more!
```
Comment (cescoffier) [inline _posts/2021-11-23-smallrye-stork-intro.adoc:59]: A demo and...

File: _posts/2021-11-23-smallrye-stork-intro.adoc [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+Take a look at the https://www.youtube.com/watch?v=l3mLKU3wR2A[70th episode of Quarkus Insights] for a more detailed explanation and examples. The demo used https://github.com/michalszynkiewicz/rest-client-reactive-stork[this project] as the client-side, each of the steps has a corresponding commit in the main branch. The backend services were instances of https://github.com/michalszynkiewicz/configurable-rest-service[this project], and the custom service discovery server implementation can be found https://github.com/michalszynkiewicz/simple-service-discovery-server[here].
```
Comment (cescoffier) [inline _posts/2021-11-23-smallrye-stork-intro.adoc:60]: Ticks around main

File: _posts/2021-11-23-smallrye-stork-intro.adoc [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+== Conclusion
+
+This post briefly introduces SmallRye Stork and its Quarkus integration. Stork provides a way to locate and select services. It's simple and customizable.
+We will cover Stork features in more detail shortly.
```
Comment (cescoffier) [inline _posts/2021-11-23-smallrye-stork-intro.adoc:65]: In more details

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: _posts/2021-11-23-smallrye-stork-intro.adoc (hunk 1 above)
Comment (michalszynkiewicz) [reply _posts/2021-11-23-smallrye-stork-intro.adoc:5]: no, they're not

File: _posts/2021-11-23-smallrye-stork-intro.adoc (hunk 4 above)
Comment (michalszynkiewicz) [reply _posts/2021-11-23-smallrye-stork-intro.adoc:36]: `You just need to configure how you want to locate and select your services. The rest will be handled by the clients directly.` ?

File: _posts/2021-11-23-smallrye-stork-intro.adoc (hunk 8 above)
Comment (michalszynkiewicz) [reply _posts/2021-11-23-smallrye-stork-intro.adoc:65]: nope, it's "in more detail" according to all pages on English I found ;)

File: _posts/2021-11-23-smallrye-stork-intro.adoc (hunk 7 above)
Comment (michalszynkiewicz) [reply _posts/2021-11-23-smallrye-stork-intro.adoc:60]: done

File: _posts/2021-11-23-smallrye-stork-intro.adoc (hunk 6 above)
Comment (michalszynkiewicz) [reply _posts/2021-11-23-smallrye-stork-intro.adoc:59]: done

File: _posts/2021-11-23-smallrye-stork-intro.adoc (hunk 5 above)
Comment (michalszynkiewicz) [reply _posts/2021-11-23-smallrye-stork-intro.adoc:44]: done

File: _posts/2021-11-23-smallrye-stork-intro.adoc (hunk 2 above)
Comment (michalszynkiewicz) [reply _posts/2021-11-23-smallrye-stork-intro.adoc]: done

**Review by cescoffier — APPROVED** [review]

**Pull request merged**